app/lib/linters/crunch_api.rb

```diff
@@ -5,34 +5,26 @@ module Linters
   # A class called Crunch
   class CrunchApi
 
-    attr_accessor :base_url
-
-    def initialize()
-      @base_url = "https://platform.42crunch.com"
+    def initialize(file:, base_url: "https://platform.42crunch.com")
+      @file = file
+      @base_url = base_url
     end
 
     # Lints an unploaded file with 42crunch.
     #
     # @param file [File] the file to be linted
     # @return [String] the raw JSON response
-    def lint_to_json(file)
-
-      # Initialize local variables
-      create_response = String.new
-      created_api_id = String.new
-      report_response = String.new
-      report_data = String.new
-      decoded_report = String.new
+    def lint_to_json
 
       # Post the file to 42Crunch according to their API v1 "Create an API (from file)" spec
       # https://www.postman.com/get-42crunch/workspace/42crunch-api/request/13761657-335bcb36-c68f-43a1-823d-741b62b55bc6
       begin
-        create_response += RestClient.post( @base_url + "/api/v1/apis",
+        create_response = RestClient.post(URI.join(@base_url, "/api/v1/apis").to_s,
         { # Body of the request
           cid: ENV['COLLECTION_ID'], # Collection id, returned by "Create a collection"
-          name: File.basename(file.path, ".*"), # API Display Name
+          name: File.basename(@file.path, ".*"), # API Display Name
```

Comment:
rather than using instance variables within this class I would declare `file` as a private `attr_reader`.

Reply:
Done.